trustmanager/keyfilestore.go

```diff
@@ -147,7 +310,44 @@ func (s *KeyMemoryStore) ExportKey(name string) ([]byte, error) {
 // ImportKey imports the private key in the encrypted bytes into the keystore
 // with the given key ID and alias.
 func (s *KeyMemoryStore) ImportKey(pemBytes []byte, alias string) error {
-	return importKey(s, s.Retriever, s.cachedKeys, alias, pemBytes)
+	err := importKey(s, s.Retriever, s.cachedKeys, alias, pemBytes)
+	if err != nil {
+		return err
+	}
+	keyID, keyInfo := getImportedKeyInfo(pemBytes, alias)
+	s.keyInfoMap[keyID] = keyInfo
+	return nil
+}
+
+func getImportedKeyInfo(pemBytes []byte, filename string) (string, KeyInfo) {
```

Comment:
Since this is now used for both generateKeyInfoMap and for importing, should this be renamed to something like keyInfoFromPEM or some such? Also, it looks like sometimes a role may not be able to be inferred if there is no role header in the PEM and the key path is not legacy-named. Should this return an error, if so? It seems like we should not be able to import any keys where we cannot infer the role (or we should write the role to the PEM headers on import).

Also previously generateKeyMap skipped keys with no role - as getImportedKeyInfo is currently used, it looks like it'd store the key IDs of keys with no role. Are we ok with that?